src/context/useSettings.tsx

```diff
@@ -4,9 +4,28 @@ import { fetchSettings } from "api/server";
 import { LxdSettings } from "types/server";
 import { UseQueryResult } from "@tanstack/react-query/src/types";
 
-export const useSettings = (): UseQueryResult<LxdSettings> => {
+type TransformedLxdSettings = {
+  [key in keyof LxdSettings]: key extends "api_extensions"
+    ? { [key: string]: boolean }
+    : LxdSettings[key];
+};
+
+export const useSettings = (): UseQueryResult<TransformedLxdSettings> => {
+  const fetchAndTransformSettings = async () => {
+    const settings = await fetchSettings();
+    const apiExtensions: TransformedLxdSettings["api_extensions"] = {};
+    settings.api_extensions?.forEach((extension) => {
+      apiExtensions[extension] = true;
+    });
+
```

Comment:
Probably better to keep this as it was and move the transformation to `useApiExtensions`, no?

Reply:
yeah 100%, also thinking changing `useApiExtensions` to `useSupportedFeatures` for better naming